packages/headless/src/features/search/search-actions-loader.ts

```diff
@@ -47,6 +48,13 @@ export interface SearchActionCreators {
     void,
     AsyncThunkSearchOptions<StateNeededByExecuteSearch>
   >;
+
+  /**
+   * Clears the search response state.
+   *
+   * @returns A dispatchable action.
+   */
+  clearSearchReponse(): PayloadAction;
```

Comment:
Why is this action needed?

Reply:
Expecting a discussion so this is great 👍
I added this action because in a salesforce community it’s all a single page app. navigating to the search page then to another page then back to the search page, we keep the engine instance alive in memory and reconnect the components to it. So far this feels pretty efficient. However, on an n>1 visit to the search page from the standalone searchbox, dispatching the `executeFirstSearchAfterStandaloneSearchBoxRedirect` no longer executes a search because firstSearchExecutedSelector  returns true so the whole execution is cancelled. So this allows me to reset it.

Reply:
What if a new engine instance was created every time the full search page loaded? Performance-wise, I would expect no noticeable difference,

Persisting the engine also means the search parameters from the initial navigation would be used (e.g page number, results per page, sort criteria, active facet values), which I don't think is right? (if it is though, we can manage this with the search parameter manager)

I feel exposing such a low-level action could become a crutch to avoid tackling certain issues.

Reply:
I agree with what you're saying. Hopefully you're right about there being no noticeable difference. I'll come up with something different that likely won't involve changes to headless.